base/linalg/triangular.jl

```diff
@@ -1291,87 +1291,262 @@ function A_rdiv_Bt!(A::StridedMatrix, B::UnitLowerTriangular)
     A
 end
 
+for f in (:A_mul_B!, :A_ldiv_B!)
+    @eval begin
+        $f(A::UpperTriangular, B::UpperTriangular) =
+            UpperTriangular($f(A, triu!(B.data)))
+        $f(A::UnitUpperTriangular, B::UpperTriangular) =
+            UpperTriangular($f(A, triu!(B.data)))
+        $f(A::UpperTriangular, B::UnitUpperTriangular) =
+            UpperTriangular($f(A, triu!(B.data)))
```

Comment:
I don't think this is correct, it uses the diagonals from `B.data` without setting them to 1 first.
```
 _/ |\__'_|_|_|\__'_|  |  Commit 1ce4ef0* (0 days old master)
|__/                   |  x86_64-w64-mingw32

julia> A = UpperTriangular(rand(5,5))
5×5 UpperTriangular{Float64,Array{Float64,2}}:
 0.817395  0.512978  0.151383  0.545986   0.680636
  ⋅        0.371017  0.326811  0.0279858  0.242921
  ⋅         ⋅        0.738791  0.0559057  0.272851
  ⋅         ⋅         ⋅        0.780668   0.0159331
  ⋅         ⋅         ⋅         ⋅         0.285675

julia> B = LinAlg.UnitUpperTriangular(rand(5,5))
5×5 Base.LinAlg.UnitUpperTriangular{Float64,Array{Float64,2}}:
 1.0  0.881164  0.0395179   0.265299  0.901799
 0.0  1.0       0.00318107  0.382759  0.909275
 0.0  0.0       1.0         0.525351  0.858504
 0.0  0.0       0.0         1.0       0.447239
 0.0  0.0       0.0         0.0       1.0

julia> B.data
5×5 Array{Float64,2}:
 0.169828  0.881164   0.0395179   0.265299  0.901799
 0.244341  0.701702   0.00318107  0.382759  0.909275
 0.952765  0.907991   0.68711     0.525351  0.858504
 0.924607  0.821667   0.173782    0.737463  0.447239
 0.22737   0.0277924  0.332617    0.311489  0.0209441

julia> C = LinAlg.A_mul_B!(A, B)
5×5 UpperTriangular{Float64,Array{Float64,2}}:
 0.138816  1.08022   0.137951  0.895375  1.59197
  ⋅        0.260344  0.225736  0.334339  0.63553
  ⋅         ⋅        0.507631  0.429353  0.664973
  ⋅         ⋅         ⋅        0.575714  0.349478
  ⋅         ⋅         ⋅         ⋅        0.0059832

julia> full(A) * full(B)
5×5 Array{Float64,2}:
 0.817395  1.39594   0.379941  1.51437   2.59939
 0.0       0.371017  0.410563  0.292349  0.705814
 0.0       0.0       0.738791  0.373108  0.783665
 0.0       0.0       0.0       0.780668  0.28876
 0.0       0.0       0.0       0.0       0.285675
```